classes/publication/Repository.inc.php

```diff
@@ -86,7 +86,11 @@ public function version(Publication $publication): int
         // Get some data about the publication being versioned before any changes are made
         $oldPublicationFormats = $publication->getData('publicationFormats');
         $oldAuthors = iterator_to_array(
-            Services::get('author')->getMany(['publicationIds' => $publication->getId()])
+            Repo::author()->getMany(
+                Repo::author()
+                    ->getCollector()
+                    ->filterByPublicationIds([$publication->getId()])
+            )
```

Comment:
You no longer need to run the `interator_to_array()` on this. It will return a `LazyCollection` which works with the `foreach` loop below where `$oldAuthors` is used.

Reply:
Fixed